packages/gatsby/src/schema/infer-graphql-type.js

```diff
@@ -153,30 +155,20 @@ function inferFromMapping(
     return null
   }
 
-  const findNode = (fieldValue, path) => {
-    const linkedNode = _.find(
-      getNodesByType(linkedType),
-      n => _.get(n, linkedField) === fieldValue
-    )
-    if (linkedNode) {
```

Comment:
after removing this check (and other checks like this in other places) we will sometimes try to add dependency to not existing node (reason for why it's failing for me currently)